src/main/java/org/folio/circulation/domain/Item.java

```diff
@@ -182,18 +182,23 @@ private boolean matchesPrimaryServicePoint(UUID servicePointId) {
   }
 
   private boolean matchesAnyServingServicePoint(UUID servicePointId) {
-    return toStream(locationRepresentation, "servicePointIds")
-      .map(UUID::fromString)
-      .anyMatch(servingServicePointId ->
-        matchingId(servicePointId, servingServicePointId));
+    return Optional.ofNullable(location)
```

Comment:
Now that a location is modelled as a class rather than JSON, could this logic be moved to the `Location` itself?

Reply:
Moved logic to the `Location` class and prevented NPEs.